nvidia_deeplearningexamples_ssd.md

```diff
@@ -0,0 +1,145 @@
+featured_image_2: ssd.png
+accelerator: cuda
+order: 10
+---
+
+
+```python
+import torch 
+precision = 'fp32'
+ssd_model = torch.hub.load('NVIDIA/DeepLearningExamples', 'nvidia_ssd', model_math=precision)
+```
+
+will load an SSD model pretrained on COCO dataset from Torch Hub.
+
+Setting precision='fp16' will load a checkpoint trained with [mixed precision](https://arxiv.org/abs/1710.03740) into architecture enabling execution on [Tensor Cores](https://developer.nvidia.com/tensor-cores).
+Handling mixed precision data requires [Apex](https://github.com/NVIDIA/apex) library.
+
+
+
+### Model Description
+
+This SSD300 model is based on the
+[SSD: Single Shot MultiBox Detector](https://arxiv.org/abs/1512.02325) paper, which
+describes SSD as “a method for detecting objects in images using a single deep neural network".
+The input size is fixed to 300x300.
+
+The main difference between this model and the one described in the paper is in the backbone.
+Specifically, the VGG model is obsolete and is replaced by the ResNet-50 model.
+
+From the
+[Speed/accuracy trade-offs for modern convolutional object detectors](https://arxiv.org/abs/1611.10012)
+paper, the following enhancements were made to the backbone:
+*   The conv5_x, avgpool, fc and softmax layers were removed from the original classification model.
+*   All strides in conv4_x are set to 1x1. 
+
+The backbone is followed by 5 additional convolutional layers.
+In addition to the convolutional layers, we attached 6 detection heads:
+*   The first detection head is attached to the last conv4_x layer.
+*   The other five detection heads are attached to the corresponding 5 additional layers.
+
+Detector heads are similar to the ones referenced in the paper, however,
+they are enhanced by additional BatchNorm layers after each convolution.
+
+### Example
+
+In the example below we will use the pretrained SSD model loaded from Torch Hub to detect objects in sample images and visualize the result.
+
+To run the example you need some extra python packages installed.
+These are needed for preprocessing images and visualization.
+
+```bash
+pip install numpy scipy scikit-image matplotlib
+```
+
+For convenient and comprehensive formatting of input and output of the model, load a set of utility methods.
+```python
+utils = torch.hub.load('NVIDIA/DeepLearningExamples', 'nvidia_ssd_processing_utils')
+```
+
+Now, prepare the loaded model for inference
+```python
+ssd_model.to('cuda')
+ssd_model.eval()
+```
+
+
+Prepare input images for object detection.
+(Example links below correspond to first few test images from the COCO dataset.)
+```python
+uris = [
+    'http://images.cocodataset.org/val2017/000000397133.jpg',
+    'http://images.cocodataset.org/val2017/000000037777.jpg',
+    'http://images.cocodataset.org/val2017/000000252219.jpg'
+]
+```
+Format the images to comply with the network input and convert them to tensor.
+```python
+inputs = [utils.prepare_input(uri) for uri in uris]
```

Comment:
can you clarify or specify (maybe just add a comment), how does one specify a local image (i.e. not a `uri`)

Reply:
yes. you can put a local path here straightforward. added a comment about it.